cairo/tests/ethereum/cancun/vm/precompiled_contracts/test_alt_bn128.py

```diff
@@ -26,10 +27,120 @@ def data_strategy(draw):
         return draw(st.binary(min_size=0, max_size=base * 193))
 
 
+@st.composite
+def add_strategy(draw):
+    """
+    Strategy for generating test cases for the alt_bn128_add precompiled contract.
+    Covers:
+    - Valid points on the ALT_BN128 curve (including point at infinity).
+    - Invalid inputs: coordinates >= ALT_BN128_PRIME or points not on the curve.
+    """
+    case_type = draw(
+        st.sampled_from(
+            [
+                "both_valid",
+                "first_infinity",
+                "second_infinity",
+                "both_infinity",
+                "out_of_range",
+                "invalid_p0",
+                "invalid_p1",
+            ]
+        )
+    )
+
+    if case_type == "both_valid":
+        p0 = AltBn128.random_point(retry=True)
+        p1 = AltBn128.random_point(retry=True)
+        return (
+            p0.x.to_bytes(32, "big")
+            + p0.y.to_bytes(32, "big")
+            + p1.x.to_bytes(32, "big")
+            + p1.y.to_bytes(32, "big")
+        )
+
+    elif case_type == "first_infinity":
+        p0 = BNP.point_at_infinity()
+        p1 = AltBn128.random_point(retry=True)
+        return (
+            p0.x.to_bytes(32, "big")
+            + p0.y.to_bytes(32, "big")
+            + p1.x.to_bytes(32, "big")
+            + p1.y.to_bytes(32, "big")
+        )
+
+    elif case_type == "both_infinity":
+        p0 = BNP.point_at_infinity()
+        p1 = BNP.point_at_infinity()
+        return (
+            p0.x.to_bytes(32, "big")
+            + p0.y.to_bytes(32, "big")
+            + p1.x.to_bytes(32, "big")
+            + p1.y.to_bytes(32, "big")
+        )
+
+    elif case_type == "second_infinity":
+        p0 = AltBn128.random_point(retry=True)
+        p1 = BNP.point_at_infinity()
+        return (
+            p0.x.to_bytes(32, "big")
+            + p0.y.to_bytes(32, "big")
+            + p1.x.to_bytes(32, "big")
+            + p1.y.to_bytes(32, "big")
+        )
+
+    elif case_type == "out_of_range":
+        # Generate coordinates, ensuring at least one is >= ALT_BN128_PRIME
+        x0 = draw(st.integers(min_value=0, max_value=ALT_BN128_PRIME + 100))
+        y0 = draw(st.integers(min_value=0, max_value=ALT_BN128_PRIME + 100))
+        x1 = draw(st.integers(min_value=0, max_value=ALT_BN128_PRIME + 100))
+        y1 = draw(st.integers(min_value=0, max_value=ALT_BN128_PRIME + 100))
+        if max(x0, y0, x1, y1) < ALT_BN128_PRIME:
+            coord = draw(st.sampled_from(["x0", "y0", "x1", "y1"]))
+            if coord == "x0":
+                x0 = ALT_BN128_PRIME + draw(st.integers(min_value=0, max_value=100))
+            elif coord == "y0":
+                y0 = ALT_BN128_PRIME + draw(st.integers(min_value=0, max_value=100))
+            elif coord == "x1":
+                x1 = ALT_BN128_PRIME + draw(st.integers(min_value=0, max_value=100))
+            else:
+                y1 = ALT_BN128_PRIME + draw(st.integers(min_value=0, max_value=100))
```

Comment:
Is there a reason to choose 100 or is it arbitrary ?
Could put it in a variable

Reply:
easier to just remove these offests, `ALT_BN128_PRIME` would be enough to trigger the condition